lib/node_modules/@stdlib/iter/cunone/lib/main.js

```diff
@@ -0,0 +1,152 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var setReadOnly = require( '@stdlib/utils-define-nonenumerable-read-only-property' );
+var isIteratorLike = require( '@stdlib/assert-is-iterator-like' );
+var isFunction = require( '@stdlib/assert-is-function' );
+var iteratorSymbol = require( '@stdlib/symbol-iterator' );
+var format = require( '@stdlib/string-format' );
+
+
+// MAIN //
+
+/**
+* Returns an iterator which cumulatively tests whether every iterated value is falsy.
+*
+* @param {Iterator} iterator - input iterator
+* @throws {TypeError} must provide an iterator
+* @returns {Iterator} iterator
+*
+* @example
+* var array2iterator = require( '@stdlib/array-to-iterator' );
+*
+* var it = iterCuNone( array2iterator( [ false, null, undefined, '', 0, NaN, true ] ) );
+* // returns <Object>
+*
+* var v = it.next().value;
+* // returns true
+*
+* v = it.next().value;
+* // returns true
+*
+* v = it.next().value;
+* // returns true
+*
+* v = it.next().value;
+* // returns true
+*
+* v = it.next().value;
+* // returns true
+*
+* v = it.next().value;
+* // returns true
+*
+* v = it.next().value;
+* // returns false
+*
+* var bool = it.next().done;
+* // returns true
+*/
+function iterCuNone( iterator ) {
+	var iter;
+	var FLG;
+	if ( !isIteratorLike( iterator ) ) {
+		throw new TypeError( format( 'invalid argument. Must provide an iterator. Value: `%s`.', iterator ) );
+	}
+	// Create a new iterator:
+	iter = {
+		'next': next
+	};
+	setReadOnly( iter, 'return', finish );
+
+	// If an environment supports `Symbol.iterator`, make the iterator iterable:
+	if ( iteratorSymbol && isFunction( iterator[ iteratorSymbol ] ) ) {
+		setReadOnly( iter, iteratorSymbol, factory );
+	}
+	return iter;
+
+	/**
+	* Returns an iterator protocol-compliant object containing the next iterated value.
+	*
+	* @private
+	* @returns {Object} iterator protocol-compliant object
+	*/
+	function next() {
+		var v;
+		if ( FLG ) {
+			return {
+				'done': true
+			};
+		}
+		v = iterator.next();
+		if ( v.done ) {
+			FLG = true;
+			return v;
+		}
+		if ( v.value ) {
+			FLG = true;
```

Comment:
I don't think this is what we want. Once this is set, the iterator ends. We want the iterator to continue iterating so long as a source iterator has values.

Reply:
Again, this is a cumulative iterator so the iterator should have the same length as the source iterator.